File: app/src/main/java/cn/intret/app/picgo/model/ViewMode.java
package cn.intret.app.picgo.model;


public enum ViewMode {
    GRID_VIEW,
    LIST_VIEW,
    UNKNOWN;

    public static ViewMode fromString(String string) {
        if (string == null) {
            return UNKNOWN;
        }
        switch (string) {
            case "GRID_VIEW": return GRID_VIEW;
            case "LIST_VIEW": return LIST_VIEW;
        }
        return UNKNOWN;
    }

    @Override
    public String toString() {
        switch (this) {
            case LIST_VIEW: return "LIST_VIEW";
            case GRID_VIEW: return "GRID_VIEW";
        }
        return "UNKNOWN";
    }
}

File: app/src/main/java/cn/intret/app/picgo/model/user/ViewMode.kt
package cn.intret.app.picgo.model.user


enum class ViewMode {
    GRID_VIEW,
    LIST_VIEW,
    UNKNOWN;

    override fun toString(): String {
        when (this) {
            LIST_VIEW -> return "LIST_VIEW"
            GRID_VIEW -> return "GRID_VIEW"
        }
        return "UNKNOWN"
    }

    companion object {

        fun fromString(string: String?): ViewMode {
            if (string == null) {
                return UNKNOWN
            }
            when (string) {
                "GRID_VIEW" -> return GRID_VIEW
                "LIST_VIEW" -> return LIST_VIEW
            }
            return UNKNOWN
        }
    }
}
